src/algorithms/synthesis/dd_synthesis.cpp

```diff
@@ -584,6 +584,38 @@ namespace syrec {
 
         extend(tt);
 
+        // Refer to the one-pass synthesis  algorithm of https://www.cda.cit.tum.de/files/eda/2017_tcad_one_pass_synthesis_reversible_circuits.pdf.
+        if (onePass) {
+            if (m > n) {
+                // zeros are inserted to match the length of the output patterns.
+                augmentWithConstants(tt, m);
+            }
+
+            // k -> Minimum no. of additional lines required (no encoding is performed).
+            const auto [_, k] = encodeHuffman(tt, true);
+
+            const auto maxIO = std::max(n, m);
+
+            r           = (m + k) - maxIO;
+            totalNoBits = maxIO + r;
+
+            // based on the totalNoBits, zeros are appended to the inputs and the outputs.
+            augmentWithConstants(tt, totalNoBits, false, true, true);
```

Comment:
why can't this `augment` call be unified with the call above? 
It should be possible to determine the number of additional lines  without the first modification.

Reply:
The first augment call (`augmentWithConstants(tt, m);`) inserts zeros to the inputs based on the **no. of primary outputs.** The second call (`augmentWithConstants(tt, totalNoBits, false, true, true)`) appends zeros to the inputs and the outputs based the **total number of bits of the circuit**.